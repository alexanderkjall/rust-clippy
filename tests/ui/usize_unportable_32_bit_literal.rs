#![warn(clippy::usize_unportable_32_bit_literal)]
#![allow(clippy::no_effect)]

fn main() {
    // assignment of value larger than 32 bit can express
    let _: usize = 4294967296;

    // single literals are also problematic
    4294967297usize;

    // A counter example
    let _: usize = 1;

    // u32::MAX shouldn't be a problem
    let _: usize = 4_294_967_295usize;

    // if the code only runs on 64 bit platforms it's not a problem
    #[cfg(target_pointer_width = "64")]
    let _: usize = 4294967296;

    // single literals on 64bit platforms only is not a problem either
    #[cfg(target_pointer_width = "64")]
    4294967297usize;
}
