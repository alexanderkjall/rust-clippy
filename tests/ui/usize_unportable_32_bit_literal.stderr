error: assignement to usize not portable to 32bit architectures
  --> tests/ui/usize_unportable_32_bit_literal.rs:6:5
   |
LL |     let _: usize = 4294967296;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use u64 as the data type if you need values larger than 32 bit
   = note: `-D clippy::usize-unportable-32-bit-literal` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::usize_unportable_32_bit_literal)]`

error: usize literals larger than u32::MAX is not portable to 32bit architectures
  --> tests/ui/usize_unportable_32_bit_literal.rs:9:5
   |
LL |     4294967297usize;
   |     ^^^^^^^^^^^^^^^
   |
   = help: use u64 as the data type if you need values larger than 32 bit

error: aborting due to 2 previous errors

